api/controllers/link-controller.ts

```diff
@@ -29,7 +30,7 @@ export const createLink = async (
     })
     .toArray();
 
-  if (result.length !== 0) return createLink(longUrl, ipAddress);
+  if (result.length !== 0) return createLink(longUrl, ipAddress, shortCode);
```

Comment:
I think the logic is flawed here 🤔 
The length of `result` would not be zero (one of the cases) if there exists another link with the same `shortCode` and trying to create a link again with the same `customCode` will lead to infinite recursion.

Reply:
@sm-sami that's a very accute observation. While this worked for a randomly generated code, it wouldn't work for the custom code. @mdfarhaan @sreyom31 can you please check and let us know?